PopulateTools/gobierto
# 435 - Mataro budgets fixes

Closes [#435](https://github.com/PopulateTools/issues/issues/435)

## :v: What does this PR do?

Fixes some translation errors (one of them due to incorrect cache key) and deletes some erroneous budget lines.

In order to fix the budget lines in production you have to run the following script from the Rails console. It deletes more records that needed, but all of the needed ones can be reimported with the budgets pipeline:

```ruby
client = ::GobiertoBudgets::SearchEngine.client
index = GobiertoBudgets::SearchEngineConfiguration::BudgetLine.index_forecast

total_deletable_results = []

#PURGE_REGEXP = /^8121\/20\d{2}\/(\d-\d{3})|(\d\d?\d?-\d-f)|(\d-\d\d?)\/.*/
PURGE_REGEXP = /^8121\/20\d{2}\/(\d-\d{1,4}|\d{1,3}-\d-f|I\d{1,2}-\d-f)\/(I|G)$/
(2009..2018).to_a.each do |year|
  terms = [{term: { ine_code: 8121 } }, { term: { year: year } } ]
  results = client.search(index: index, type: "economic", body: { query: { filtered: { filter: { bool: { must: terms }}}}, size: 10_000 })["hits"]["hits"]
  deletable_results = results.select { |r| r["_id"] =~ PURGE_REGEXP }
  puts "Found #{deletable_results.size} deletable results for year #{year}"
  total_deletable_results += deletable_results
end

puts total_deletable_results.size

puts total_deletable_results.map{|r| r["_id"]}
````

The previous step prints the IDs of all the budget lines that will be deleted. Please review them (pay special attention that no data from dependent entities is present) before proceeding to the next step, which deletes them:

```ruby
deleted_count = 0
total_deletable_results.each do |result|
  client.delete(index: index, type: "economic", id: result["_id"])
  deleted_count += 1
  puts "Deleted #{deleted_count}/#{total_deletable_results.size} docs"
end
```

After this we must:

1. Clear rails cache
2. Re-run the jenkins budgets pipeline and check the deleted budget lines are not restored
3. (Maybe) restart ES

## How to test this?

In staging, check that funcitonal budget lines of levels 1 to 3 don't display duplicated economic budget lines. You can check this ones, or others:

* http://mataro.gobify.net/presupuestos/partidas/1/2018/functional/G
* http://mataro.gobify.net/presupuestos/partidas/15/2018/functional/G
* http://mataro.gobify.net/presupuestos/partidas/152/2018/functional/G

Compare results against production ones, and check no numbers have been broken (in any case, the pipelines can be re-executed).
